Block and Inline: Enhance explanation of div and span elements (#30638)
TheOdinProject/curriculum: foundations/html_css/css_foundations/block_and_inline.md

### Divs and spans

When building a web page, not every piece of content has a specific semantic tag. Sometimes, you just need a container you can style or position! That's where `div` and `span` come in.

> A `div` is a generic block-level container. It behaves like a rectangular section that stretches across the full width of its parent and starts on a new line. Typically, you would use a `div` to group related elements inside page components like for example cards, sections, sidebars, or navigation bars.

Reviewer: Slight grammar tweak but also specifying "by default" because divs don't *have* to start on new lines (e.g. children in a row flex container) nor do they *have* to take the full width of the parent (can override width using CSS, or flexbox aligments for example). Being clear about this just being default behaviour saves us from any "well actually" shenanigans.

```suggestion
A `div` is a generic block-level container. It behaves like a rectangular section that, by default, stretches across the full width of its parent and starts on a new line. Typically, you would use a `div` to group related elements inside page components, such as grouping cards, sections, sidebars, or navigation bars.
```

<script async src="https://cpwebassets.codepen.io/assets/embed/ei.js"></script>

A `span` is a generic **inline** container. It sits inside a line of text and only takes up as much space as its content. Unlike `div`, it does **not** start on a new line. You use `span` when you want to style or target just part of a sentence.

> For example, highlighting a single word or attaching a tooltip to a phrase. Again, you add a `class` or `id` so CSS can find and manipulate it.

Reviewer: It doesn't read as well with the split paragraph, partly because "For example, highlighting a single word or attaching a tooltip to a phrase." doesn't make sense as a standalone sentence.

Keeping the two paragraphs as one and joining the "for example" sentence with the previous makes more sense and is easier to read.

> For example, highlighting a single word or attaching a tooltip to a phrase. Again, you add a `class` or `id` so CSS can find and manipulate it.

Reviewer 2: @dayview I didn't realise this was still open, but yes, reverting that would be great, that would be all that's needed to complete this in my opinion.

> For example, when you want to highlight a single word or attaching a tooltip to a phrase you can add a `class` or `id` so CSS can find and manipulate it.

Reviewer: Grammar nit
```suggestion
A `span` is a generic inline container. It sits inside a line of text and only takes up as much space as its content. Unlike `div`, it does not start on a new line. You use `span` when you want to style or target just part of a sentence. For example, when you want to highlight a single word or attach a tooltip to a phrase, you can wrap it in a `span` then add a `class` or `id` so CSS can find and manipulate it.
```

<script async src="https://cpwebassets.codepen.io/assets/embed/ei.js"></script>

Neither `div` nor `span` adds meaning to the content the way semantic tags do, as they exist to give you flexible "building blocks". The key difference is how they behave in the layout: `div` participates in the flow as a block-level element, while `span` behaves as an inline element.

> Once you understand that, choosing between them becomes simple! You can use `div` when you're grouping and positioning bigger blocks of content, and `span` when you're styling or scripting small pieces inside a line.

Reviewer 2: I also think this can be one paragraph.
